src/test/java/net/imagej/convert/ConvertROITreeToMaskTest.java

```diff
@@ -0,0 +1,234 @@
+				)), //
+				DefaultWritablePolyline.class, //
+			}, //
+			new Object[] { //
+				new RealMask()
+				{
+
+					@Override
+					public boolean test(RealLocalizable realLocalizable) {
+						return realLocalizable.getDoublePosition(0) + realLocalizable
+							.getDoublePosition(1) == 0;
+					}
+
+					@Override
+					public int numDimensions() {
+						return 2;
+					}
+				}, //
+				RealMask.class //
+			}, //
+			new Object[] { //
+				new MaskPredicate<RealLocalizable>()
+				{
+
+					@Override
+					public int numDimensions() {
+						return 2;
+					}
+
+					@Override
+					public boolean test(RealLocalizable realLocalizable) {
+						// true iff any coordinate is zero
+						return Arrays //
+							.stream(realLocalizable.positionAsDoubleArray()) //
+							.anyMatch(d -> d == 0);
+					}
+
+					@Override
+					public MaskPredicate<RealLocalizable> and(
+						Predicate<? super RealLocalizable> other)
+				{
+						throw new UnsupportedOperationException();
+					}
+
+					@Override
+					public MaskPredicate<RealLocalizable> or(
+						Predicate<? super RealLocalizable> other)
+				{
+						throw new UnsupportedOperationException();
+					}
+
+					@Override
+					public MaskPredicate<RealLocalizable> negate() {
+						throw new UnsupportedOperationException();
+					}
+
+					@Override
+					public MaskPredicate<RealLocalizable> minus(
+						Predicate<? super RealLocalizable> other)
+				{
+						throw new UnsupportedOperationException();
+					}
+
+					@Override
+					public MaskPredicate<RealLocalizable> xor(
+						Predicate<? super RealLocalizable> other)
+				{
+						throw new UnsupportedOperationException();
+					}
+				}, //
+				MaskPredicate.class //
+			} //
+		);
+	}
+
+	@Test
+	public void testROITreeToROITest() {
+		// Construct the ROI tree
+		ROITree tree = new DefaultROITree();
+		tree.addROIs(Collections.singletonList(ROI));
+		// Assert that the convertService can convert this tree to a mask
+		final ConvertService convertService = c.service(ConvertService.class);
+		MaskPredicate<?> actual = convertService.convert(tree, ROIClass);
+		Assert.assertEquals(ROI, actual);
+	}
+
+	@Test
+	public void testROIToROITreeTest() {
+		// Assert that the convertService can convert this mask to a tree
+		final ConvertService convertService = c.service(ConvertService.class);
+		ROITree actual = convertService.convert(ROI, ROITree.class);
+		Assert.assertNotNull(actual);
+		List<TreeNode<?>> rois = actual.children();
+		Assert.assertEquals(1, rois.size());
+		// Assert roi equality
+		Assert.assertEquals(ROI, rois.get(0).data());
+	}
+
+	@Test
+	public void testTooFullROITree() {
+		// Construct the ROI tree
+		ROITree tree = new DefaultROITree();
+		tree.addROIs(Arrays.asList(ROI, ROI));
+		// Assert that the convertService cannot convert this tree to a mask
+		final ConvertService convertService = c.service(ConvertService.class);
+		Assert.assertNull(convertService.convert(tree, ROI.getClass()));
```

Comment:
Similar to above, we can use `convertService.supports()` here.

```suggestion
		final ConvertService convertService = c.service(ConvertService.class);
		Assert.assertFalse(convertService.supports(tree, ROI.getClass()));
```

Reply:
Also fixed in e6ed111